### PR TITLE
Support Content-Security-Policy

### LINKED ISSUE #2845: Support strict CSP
Icinga Web 2.12 introduced the ability to enable strict content security policy that the Director does not support yet. Inline JS must be avoided. Inline CSS should be avoided or protected by a nonce.

### PR DESCRIPTION
**Pull request by sukhwinder33445**
resolves #2845

### CODE REVIEW

File: public/css/module.less [hunk 1]
```diff
@@ -1833,3 +1872,38 @@ table.table-basket-changes {
...
+    color: green;
+  }
+}
+
+.seviceset-obj-link {
+  float: right;
+  font-weight: normal
+}
+
+//DataController::dictionaryAction, cannot findusage in web
```
Comment (raviks789) [inline public/css/module.less:1902]: You can remove this comment, I have tested this.

File: application/forms/AddToBasketForm.php [hunk 2]
```diff
@@ -41,21 +38,14 @@ public function setup()
...
-                $count++;
+            if (! $basket || ! $basket->hasObject($type, $name)) {
                 $names[] = $name;
+                $count++;
             }
         }
         $this->addHtmlHint((new HtmlDocument())->add([
             'The following objects will be added: ',
-            $names
+            (array) implode(", ", $names)
```
Comment (raviks789) [inline application/forms/AddToBasketForm.php:48]: instead of casting the string to array could you not perform string concatenation?

**Review by nilmerg — CHANGES_REQUESTED** [review]
There's still an onclick attribute defined in `ImportsourceController:183`

**Review by nilmerg — APPROVED** [review]

**Comment by nilmerg** [comment]
please rebase

**Pull request merged**